### ISSUE TITLE
Waiting for local host forever when I am trying deploy Flask API using wsgi Mod in apache24

### ISSUE THREAD

**Issue by murthy0795** [GrahamDumpleton/mod_wsgi#532]
Waiting for local host for hours when I am trying deploy Flask API using wsgi Mod in apache24...Below was my code
_________________________________________________________
from flask import Flask                                     # to deploy as a flask service
from flask import request                                   # to be able to read the incoming requests
from flask import jsonify                                   # to parse the response as JSON
from flask import abort                                     # to close the connection with the client with an error code
from flask_cors import CORS                                 # to enable Cross-Origin Resource Sharing
from flask import make_response
import logging
import logging.config                                              # to perform logging
import warnings
from io import StringIO
from config1 import Config1
import tensorflow

import os
from tensorflow.keras.preprocessing import image
from tensorflow.keras.models import load_model

from pillow import Image
#import pytesseract
import argparse
import cv2
import re
import io
import json
import ftfy
import pandas as pd
import numpy as np
import pytesseract as pt

#from doc_classification_similarity import test_similarity,test_model_classification

warnings.filterwarnings("ignore", message="numpy.dtype size changed")
warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
###########################################End of Import Statements#####################################################
# Initializing Flask Application
app = Flask(__name__)                                       # 'app' refers to initialized Flask App
app.config["DEBUG"] = True
CORS(app)                                                   # to enable CORS for the services
app.config.from_object(Config1)
#app.config.from_object('config')
app.config.from_object('config1.Config1')
                                               # attaching the configurations to the flask app

# Configuring Logs
root_logger = logging.getLogger()
root_logger.setLevel(Config1.LOG_LEVEL)
handler = logging.FileHandler(Config1.LOG_FILENAME, 'a+', 'utf-8')
handler.setFormatter(logging.Formatter(fmt=Config1.LOG_FORMAT, datefmt=Config1.LOG_DATE_FORMAT))
root_logger.addHandler(handler)

# Declaration and definition of the Web Service
@app.route('/kyc_class' ,methods = ['POST'])
def doc_class():
    
    # Declaration of variables
    result = {}
         
    # If file is not received from UI end
    if 'file' not in request.files:
        result["Status Code"] = "500"
        result["Message"] = "Data File not received as Expected"
        return jsonify(result)
    else:
        data_file = request.files['file']
        logging.info("File Recevied {0}".format(data_file.filename))
        filename = data_file.filename.split('.')
        file_location = os.path.join(Config1.OUTPUT_LOCATION, filename[0] + '.' + filename[1])
        print("filelocation=",file_location)
        try:
            data_file = request.files['file'].save(file_location)
        except Exception as e:
            result["Status Code"] = "501"
            result["Message"] = "File is not Uploaded: " + str(e)
            return jsonify(result)
        logging.info("Data file is uploaded at output/received")

    logging.info("Classification for Dataset")
    try:
        test_data=os.path.join(Config1.OUTPUT_DIRECTORY,file_location)
        model=os.path.join(Config1.MODEL_DIRECTORY,Config1.MODEL_CLASSIFICATION)
        classifier = load_model(model)
       
        kyc_test_image = image.load_img(test_data, target_size = (64, 64))
        kyc_test_image = image.img_to_array(kyc_test_image)
        kyc_test_image = np.expand_dims(kyc_test_image, axis = 0)
                
        outcome = classifier.predict(kyc_test_image)
        if int(outcome[0][2]) == 1:
            pred='Pancard Uploaded'
        else:
            pred='Invalid document, Please upload a valid Pancard'
            result["Message"] = pred
            return jsonify(result)

       
        print("classification prediction=",pred)
    except Exception as e:
        result["Status Code"] = "505"
        result["Message"] = "Error invalid document uploaded: " + str(e)
        return jsonify(result)
    
    logging.info("Pan Number Extraction")
    
    try:
                    
        image1 = cv2.imread(test_data)
        #filename = "C:\\Users\\Desktop\\Files\\CNN_nk\\Pan text.jpg".format(os.getpid())
        image1 = cv2.resize(image1, None, fx=2.75, fy=2.5, interpolation=cv2.INTER_CUBIC)
        
        gray = cv2.cvtColor(image1, cv2.COLOR_BGR2GRAY)

        img = gray = cv2.bilateralFilter(gray, 9, 75, 75)
        #gray = cv2.bitwise_not(gray)
        img = cv2.threshold(gray, 0, 255, cv2.THRESH_BINARY | cv2.THRESH_OTSU)[1]
        
        #img = gray = cv2.bilateralFilter(gray, 9, 75, 75)
        
        kernel = np.ones((2, 1), np.uint8)
        img = cv2.dilate(img, kernel, iterations=2)
        img = cv2.erode(img, kernel, iterations=1)

        #img = gray = cv2.bilateralFilter(gray, 9, 75, 75)
        
        #cv2.imwrite(filename, img)

        text = pt.image_to_string(img, lang = 'eng')
  
        text.split('\n')
        
        m = re.search(r'\b([a-zA-Z]){5}([0-9]){4}([a-zA-Z]){1}\b', text)
        if m is not None:
            pan = m.group()
            print(pan)
        else:
            pan = "Invalid image, please upload a different image"
       
        result["Pancard Id"]=pan
        result["document type"]=pred
        return jsonify(result)
    
    except Exception as e:
        result["Status Code"] = "505"
        result["Message"] = "Invalid document error: " + str(e)
        return jsonify(result)

#app.run(host="0.0.0.0",port=8089, threaded=True)
_______________________________________
httpd.vhost
_____________________________________
<VirtualHost *:8089>
 	ServerName localhost
 	WSGIScriptAlias / "C:/kyc_cnn_v1/app/web.wsgi"
 	DocumentRoot C:/kyc_cnn_v1/app
 	<Directory "C:/kyc_cnn_v1/app/">
 	Options +Indexes +Includes +FollowSymLinks +MultiViews +ExecCGI
 	AllowOverride All
 	Require local
 	</Directory>
</VirtualHost>
__________________________________________________________________________
Error Logs
__________________________________________________________________________
[Wed Feb 26 18:52:28.770149 2020] [mpm_winnt:notice] [pid 5720:tid 760] AH00455: Apache/2.4.41 (Win64) mod_wsgi/4.7.0 Python/3.7 configured -- resuming normal operations
[Wed Feb 26 18:52:28.771146 2020] [mpm_winnt:notice] [pid 5720:tid 760] AH00456: Apache Lounge VS16 Server built: Aug  9 2019 16:46:32
[Wed Feb 26 18:52:28.771146 2020] [core:notice] [pid 5720:tid 760] AH00094: Command line: 'httpd.exe -d C:/Apache24'
[Wed Feb 26 18:52:28.811191 2020] [mpm_winnt:notice] [pid 5720:tid 760] AH00418: Parent: Created child process 10744
[Wed Feb 26 18:52:29.431563 2020] [mpm_winnt:notice] [pid 10744:tid 840] AH00354: Child: Starting 64 worker threads.

__________________________________________________________
Please help me in resolving the issue

**Comment by GrahamDumpleton** [comment]
It is unclear what you are saying the problem is. If you are saying that the web request never returns, it is likely because the packages you are using aren't designed to work in Python sub interpreters. This is an issue with packages it data science space.

To resolve it if it is this issue, you need to force the use of the Python main interpreter. See:

* https://modwsgi.readthedocs.io/en/develop/user-guides/application-issues.html#python-simplified-gil-state-api

Specifically, add:

```
WSGIApplicationGroup %{GLOBAL}
```

**Comment by GrahamDumpleton** [comment]
Closing as no further response and assumed information provided was sufficient to solve any issue or gave up.
